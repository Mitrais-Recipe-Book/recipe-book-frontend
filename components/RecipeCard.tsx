import React from "react";
import Image from "next/image";
import { SiCodechef } from "react-icons/si";

// @ts-ignore
export default function RecipeCard(props) {
  // console.log(props);
  return (
    <div className="mx-2 my-3 sm:w-40 xl:w-50 box-border border-1 pb-2 rounded shadow transition-all hover:bg-orange-200 hover:scale-110">
      <Image
        className="w-full rounded-t cursor-pointer"
        src={props.recipe.recipeImage ? props.recipe.recipeImage : "/images/bibimbap-image.webp"}
        alt="RecipyBook"
        width={200}
        height={130}
        objectFit="cover"
      />
      <div className="px-2 py-1">
        <div className="font-bold text-lg cursor-pointer break-words">
          {props.recipe.recipeName}
        </div>
        <div className="text-gray-600 md:text-ellipsis">
          {props.recipe.description}
        </div>
        {props.recipe.author? (<div className="flex py-2">
          <div className="py-2 cursor-pointer ml-1 mr-2">
            {props.recipe.authorImage ? <Image src={props.recipe.authorImage} className="rounded-full" width={24} height={24} objectFit="cover" alt="author profile" /> : <SiCodechef className="rounded-full" size={24} />}
          </div>
          <div>
            <div className="font-semibold text-sm cursor-pointer">
              {props.recipe.author}
            </div>
            <div className="text-xs pl-1 text-gray-600 cursor-pointer">
              {props.recipe.authorFollower}
            </div>
          </div>
        </div>): null}
        
      </div>
      <div className="text-center pt-1">
        <div className="text-sm text-gray-600"> {props.recipe.recipeViews} </div>
      </div>
    </div>
  );
}
