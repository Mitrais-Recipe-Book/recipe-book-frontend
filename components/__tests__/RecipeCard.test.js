import RecipeCard from "../RecipeCard";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

describe("Test RecipeCard on index", () => {
  const recipe = {
    id: 1,
    recipeName: "Es Teh Anget",
    description: "Enak diminum untuk buka puasa",
    recipeViews: 5,
    author: {
      username: "user1",
      fullName: "User 1",
      authorFollowers: 5,
    },
  };

  it("render without crashing", () => {
    render(<RecipeCard recipe={recipe} />);
  });
  it("has recipe name", () => {
    render(<RecipeCard recipe={recipe} />);
    const linkElement = screen.getByText(/Es Teh Anget/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("has recipe description", () => {
    render(<RecipeCard recipe={recipe} />);
    const linkElement = screen.getByText(/Enak diminum untuk buka puasa/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("has recipe author", () => {
    render(<RecipeCard recipe={recipe} />);
    const linkElement = screen.getByText(/User 1/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("try to load banner image", () => {
    render(<RecipeCard recipe={recipe} />);
    const linkElement = screen.getByText(/Loading Banner Image.../i);
    expect(linkElement).toBeInTheDocument();
  });
  it("try to load profile image", () => {
    render(<RecipeCard recipe={recipe} />);
    const linkElement = screen.getByText(/Loading Profile Image.../i);
    expect(linkElement).toBeInTheDocument();
  });
  it("match the snapshot", () => {
    const tree = renderer.create(<RecipeCard recipe={recipe} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Test RecipeCard on recipe detail", () => {
  const recipe = {
    id: 1,
    recipeName: "Es Teh Anget",
    description: "Enak diminum untuk buka puasa",
  };
  it("render without crashing", () => {
    render(<RecipeCard recipe={recipe} />);
  });
  it("has recipe name", () => {
    render(<RecipeCard recipe={recipe} />);
    const linkElement = screen.getByText(/Es Teh Anget/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("has recipe description", () => {
    render(<RecipeCard recipe={recipe} />);
    const linkElement = screen.getByText(/Enak diminum untuk buka puasa/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("don't have recipe author", () => {
    render(<RecipeCard recipe={recipe} />);
    const linkElement = screen.queryByText(/User 1/i);
    expect(linkElement).not.toBeInTheDocument();
  });
  it("match the snapshot", () => {
    const tree = renderer.create(<RecipeCard recipe={recipe} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
