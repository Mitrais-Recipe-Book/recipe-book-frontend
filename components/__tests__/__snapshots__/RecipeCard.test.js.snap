// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RecipeCard on index match the snapshot 1`] = `
<div
  className="grid grid-flow-row grid-flow-row-col grid-rows-3 grid-cols-1 mx-2 my-3 sm:w-40 xl:w-50 box-border border-1 pb-2 rounded shadow transition-all hover:bg-orange-200 hover:scale-110"
>
  <section>
    <div
      className="flex self-center"
    >
      Loading Banner Image...
    </div>
  </section>
  <section
    className="mx-2 my-1 row-span-1"
  >
    <div
      className="font-bold text-lg cursor-pointer break-words line-clamp-2"
      onClick={[Function]}
      title="Es Teh Anget"
    >
      Es Teh Anget
    </div>
    <div
      className="text-gray-600 break-words line-clamp-3"
      title="Enak diminum untuk buka puasa"
    >
      Enak diminum untuk buka puasa
    </div>
  </section>
  <section
    className="object-bottom"
  >
    <div
      className="grid grid-cols-4 my-4"
    >
      <div
        className="w-3/4 m-auto"
      >
        <div>
          Loading Profile Image...
        </div>
      </div>
      <div
        className="col-span-3"
      >
        <div
          className="font-semibold text-sm cursor-pointer"
        >
          User 1
        </div>
        <div
          className="text-xs pl-1 text-gray-600 cursor-pointer"
        >
          5
        </div>
      </div>
    </div>
    <div
      className="text-center pt-1"
    >
      <div
        className="text-sm text-gray-600"
      >
         
        5
         
      </div>
    </div>
  </section>
</div>
`;

exports[`Test RecipeCard on recipe detail match the snapshot 1`] = `
<div
  className="flex-wrap mx-2 my-3 sm:w-40 xl:w-50 box-border border-1 pb-2 rounded shadow transition-all hover:bg-orange-200 hover:scale-110"
>
  <section>
    <div
      className="flex self-center"
    >
      Loading Banner Image...
    </div>
  </section>
  <section
    className="mx-2 my-1 row-span-2"
  >
    <div
      className="font-bold text-lg cursor-pointer break-words line-clamp-2"
      onClick={[Function]}
      title="Es Teh Anget"
    >
      Es Teh Anget
    </div>
    <div
      className="text-gray-600 break-words line-clamp-3"
      title="Enak diminum untuk buka puasa"
    >
      Enak diminum untuk buka puasa
    </div>
  </section>
</div>
`;
